workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Linq;
+using Improbable.Gdk.Core.Commands;
+using Unity.Collections;
+using Unity.Entities;
+
+#region Diagnostic control
+
+// Disable the "variable is never assigned" for injected fields.
+#pragma warning disable 649
+
+// ReSharper disable ClassNeverInstantiated.Global
+
+#endregion
+
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    public class GameObjectWorldCommandSystem : ComponentSystem
+    {
+        private struct ReserveEntityIdsResponseData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<WorldCommands.ReserveEntityIds.CommandResponses> CommandResponses;
+            [ReadOnly] public ComponentArray<GameObjectReference> HasGameObjectReference;
+        }
+
+        private struct CreateEntityResponseData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<WorldCommands.CreateEntity.CommandResponses> CommandResponses;
+            [ReadOnly] public ComponentArray<GameObjectReference> HasGameObjectReference;
+        }
+
+        private struct DeleteEntityResponseData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<WorldCommands.DeleteEntity.CommandResponses> CommandResponses;
+            [ReadOnly] public ComponentArray<GameObjectReference> HasGameObjectReference;
+        }
+
+        [Inject] private GameObjectDispatcherSystem gameObjectDispatcherSystem;
+        [Inject] private ReserveEntityIdsResponseData reserveEntityIdsResponseData;
+        [Inject] private CreateEntityResponseData createEntityResponseData;
+        [Inject] private DeleteEntityResponseData deleteEntityResponseData;
+
+        private static readonly InjectableId WorldCommandResponseHandlerInjectableId =
+            new InjectableId(InjectableType.WorldCommandResponseHandler, InjectableId.NullComponentId);
+
+        protected override void OnUpdate()
+        {
+            for (var i = 0; i < reserveEntityIdsResponseData.Length; ++i)
+            {
+                var entity = reserveEntityIdsResponseData.Entities[i];
+                var worldCommandResponseHandlers = GetWorldCommandResponseHandlersForEntity(entity);
+
+                if (worldCommandResponseHandlers == null)
+                {
+                    continue;
```

Comment:
can that actually happen? if it has a GameObjectReference, shouldn't it also be in the storage?

Reply:
If the entity has none injected I think it may be null... I'll do a little test though

Reply:
aye, it is null (also we have a `count==0?null` in the method that returns it too